common/error_test.cpp

```diff
@@ -203,12 +205,13 @@ TYPED_TEST(ErrorOrTest, AssignOrReturnNoErrorAcrossErrorTypes) {
 }
 
 TYPED_TEST(ErrorOrTest, AssignOrReturnErrorAcrossErrorTypes) {
-  using TestErrorOr = typename TestFixture::template TestErrorOr<int>;
+  using TestErrorOr = typename TestFixture::template TestErrorOrTemp<int>;
   auto result = []() -> ErrorOr<int> {
     CARBON_ASSIGN_OR_RETURN(int a, TestErrorOr(1));
     CARBON_ASSIGN_OR_RETURN(const int b, []() -> TestErrorOr {
-      CARBON_ASSIGN_OR_RETURN(
-          int inner, (ErrorOr<int, AnotherCustomError>)(AnotherCustomError()));
+      CARBON_ASSIGN_OR_RETURN(int inner,
+                              (static_cast<ErrorOr<int, AnotherCustomError>>(
+                                  AnotherCustomError())));
```

Comment:
```suggestion
                              (ErrorOr<int, AnotherCustomError>(
                                  AnotherCustomError()));
```

Similar to the above